v5.0.0

donavanbecker opened the pull request:
## [Version 5.0.0](https://github.com/donavanbecker/homebridge-meross/compare/v4.0.0....5.0.0) (2021-03-14)

### Breaking Changes

- The Plugin has been been changed from an `accessory` type to a `platform` type.
- You will have to change you config completely if you update to this version.
  - You can take your current `accessory` and move it to the platform config.
  - See (Specific Model Configurations)[https://github.com/donavanbecker/homebridge-meross/wiki/Specific-Model-Configurations] Wiki for more examples.
  - Example: 
   #### Before:
    ```json
    "accessories": [
        {
          "model": "MSS620",
          "name": "Outlet",
          "deviceUrl": "192.168.1.1",
          "channel": 0,
          "messageId": "abcdefghijklmnopqrstuvwxyz123456789",
          "timestamp": 123456789,
          "sign": "abcdefghijklmnopqrstuvwxyz123456789",
          "accessory": "Meross"
        }
    ]
    ```
   #### After:
    ```json
    "platforms": [
        {
        "name": "Meross",
        "devices": [
            <This_is_from_above>
            {
            "model": "MSS620",
            "name": "Outlet",
            "deviceUrl": "192.168.1.1",
            "channel": 0,
            "messageId": "abcdefghijklmnopqrstuvwxyz123456789",
            "timestamp": 123456789,
            "sign": "abcdefghijklmnopqrstuvwxyz123456789",
            "accessory": "Meross" <You_Can_Remove_This.>
            }
            <Ends_here_from_above>
          ],
        "platform": "Meross"
        }
    ]
    ```
- Added Config for Refresh Rate.
  - default is 5 seconds and if updating to often can be set in the config.

 
## Testers
Need Beta testers to try this and let me know if you have any issues and post here which model you are using and just say you have an issue, then open your own separate issue to try that model and Beta v5.0.0.

Let me know if you have any questions on updating to this beta. Easiest way is to use Homebridge UI.

Beta: `4.0.2-beta.1` on `npm`

## Tested Models:
* [ ] MSS110-1
* [x] MSS110-2
* [ ] MSL-100
* [ ] MSL-120
* [ ] MSL-320
* [x] MSL-420
* [x] MSS210
* [x] MSS310
* [ ] MSS420F
* [ ] MSS425
* [x] MSS425E
* [x] MSS425F
* [x] MSS510
* [ ] MSS510M
* [ ] MSS530H
* [ ] MSS550
* [x] MSS560
* [ ] MSS570
* [ ] MSS570x
* [ ] MSS5X0
* [x] MSS620
* [x] MSS630
* [ ] MSS710
* [x] MSG100
* [ ] MSG200

donavanbecker committed:
Update index.ts

donavanbecker committed:
breaking changes

donavanbecker committed:
v5.0.0 Beta

## [Beta Version 5.0.0](https://github.com/donavanbecker/homebridge-meross/compare/v4.0.0....5.0.0) (2021-XX-XX)

### Breaking Changes

- The Plugin has been been changed from an `accessory` type to a `platform` type.
- You will have to change you config completely if you update to this version.
  - You can take your current `accessory` and move it to the platform config.
  - Example:
    - `Before`:
    ```json
    {
      "model": "MSS620",
      "name": "Outlet",
      "deviceUrl": "192.168.1.1",
      "channel": 0,
      "messageId": "abcdefghijklmnopqrstuvwxyz123456789",
      "timestamp": 123456789,
      "sign": "abcdefghijklmnopqrstuvwxyz123456789",
      "accessory": "Meross"
    }
    ```
    - `After`:
    ```json
    {
    "name": "Meross",
    "devices": [
        // This is the from above
        {
        "model": "MSS620",
        "name": "Outlet",
        "deviceUrl": "192.168.1.1",
        "channel": 0,
        "messageId": "abcdefghijklmnopqrstuvwxyz123456789",
        "timestamp": 123456789,
        "sign": "abcdefghijklmnopqrstuvwxyz123456789",
        "accessory": "Meross" //Remove this.
        }
        // Ends here from above
      ],
    "platform": "Meross"
    }
    ```

donavanbecker commented:
@CocoaBob @dylanfrankcom @Ponyboy47 if you could test this. and make any pull requests to get this in better shape. I have did what I think needs to be done, but I only have 1 device....

donavanbecker commented:
@MrJer, I saw you tested and it works.if there is any changes that need to be made let me know.

donavanbecker committed:
remove request

donavanbecker committed:
seperate models out by device type

123marvin123 commented:
I tested using `4.0.1-beta.4`:

- The configuration ui where you can add new devices still generates the old configuration type in the "accessories" instead of "platforms" section -> new users will be confused
- MSL-420 actually does not turn off anymore. You can turn it on and change color but not off. I will investigate this issue further now.

123marvin123 commented on src/devices/lightbulbs.ts:
```diff
@@ -0,0 +1,546 @@
+      data: this.Data,
+    },
+    );
+    if (this.On) {
```
What exactly does this if statement do, why is it necessary? If I remove it, my MSL-420 works fine again.

donavanbecker replied:
weird because all that does is put text in the log to say either that it has been turned on or off.

123marvin123 replied:
I get the following log output in the homebridge console (using the code from 4.0.1-beta4):

**Turning the bulb on (1 time):**

```
[3/4/2021, 8:07:18 PM] [Meross] Sending request On for BULB
[3/4/2021, 8:07:18 PM] [Meross] Sending request undefined for BULB
[3/4/2021, 8:07:18 PM] [Meross] Sending request On for BULB
[3/4/2021, 8:07:18 PM] [Meross] Sending request On for BULB
```

**Trying to turn it back off (1 time):**

```
[3/4/2021, 8:08:56 PM] [Meross] Sending request Off for BULB
[3/4/2021, 8:08:56 PM] [Meross] Sending request undefined for BULB
[3/4/2021, 8:08:56 PM] [Meross] Sending request Off for BULB
[3/4/2021, 8:08:56 PM] [Meross] Sending request Off for BULB
```

I don't know why, but it did turn off this time, although I did not change anything in the code this time 🤨 But it's clear that something is going, where the request is sent multiple times.

123marvin123 replied:
When I select a different color, it's flashing in some other color before it's using the color I selected

donavanbecker reviewed (commented):
is it potential sending two `On` or two `Off` commands?

would you be will to submit a PR for a fix on the lightbulb devices? all I have it outlets.

123marvin123 commented:
Yes, I'll take a look at the issue.

123marvin123 commented on src/devices/lightbulbs.ts:
```diff
@@ -0,0 +1,546 @@
+    this.platform.log.debug('%s %s - Set On: %s', this.device.model, this.accessory.displayName, value);
+
+    this.On = value;
+    this.doUpdate.next();
```
Your doUpdate() get's called here which causes an update of the brightness to. That's why we are getting 4 log messages even though we are only updating the "On"-state. Also, because your are using "this.Request = ..." and calling the update methods async, you are overriding the value. That's why we are getting 4x the same log message.

123marvin123 replied:
Regarding the async stuff and overriding the value:

- That was not correct, the log messages were just wrong (copy pasted?).

123marvin123 commented:
Issues fixed with PR #230

123marvin123 committed:
Fix lightbulb issues and logging messages (#230)

123marvin123 committed:
Remove doUpdate & add proper error handling (#231)

* Fix lightbulb issues and logging messages

* remove doUpdate() and add proper error handling

* Remove UpdateInProgress variable

donavanbecker committed:
changed let to const

donavanbecker commented:
> Issues fixed with PR #230

okay I just pushed out 1 more little change to the `lightbulb.ts`, lets test it out and see how good it works.

donavanbecker committed:
Update lightbulbs.ts

donavanbecker commented:
@123marvin123 what models do you have that you have tested so I can mark them off.

123marvin123 commented:
> @123marvin123 what models do you have that you have tested so I can mark them off.

Only the MSL-420

donavanbecker committed:
add HapStatusError

donavanbecker committed:
add default values so they are not undefined.

donavanbecker committed:
update dependencies

donavanbecker committed:
Update dependabot.yml

donavanbecker committed:
add MSS570x

donavanbecker committed:
garagedoor payload fix

donavanbecker committed:
next try.

donavanbecker committed:
update dependencies

mir597 committed:
Fixed a bug that does not properly retrieve a device status. (#243)

tuemeister committed:
Removed a double quote in property (#246)

donavanbecker committed:
v5.0.0

## [Version 5.0.0](https://github.com/donavanbecker/homebridge-meross/compare/v4.0.0....5.0.0) (2021-03-14)

### Breaking Changes

- The Plugin has been been changed from an `accessory` type to a `platform` type.
- You will have to change you config completely if you update to this version.
  - You can take your current `accessory` and move it to the platform config.
  - See (Specific Model Configurations)[https://github.com/donavanbecker/homebridge-meross/wiki/Specific-Model-Configurations] Wiki for more examples.
  - Example:
   #### Before:
    ```json
    "accessories": [
        {
          "model": "MSS620",
          "name": "Outlet",
          "deviceUrl": "192.168.1.1",
          "channel": 0,
          "messageId": "abcdefghijklmnopqrstuvwxyz123456789",
          "timestamp": 123456789,
          "sign": "abcdefghijklmnopqrstuvwxyz123456789",
          "accessory": "Meross"
        }
    ]
    ```
   #### After:
    ```json
    "platforms": [
        {
        "name": "Meross",
        "devices": [
            <This_is_from_above>
            {
            "model": "MSS620",
            "name": "Outlet",
            "deviceUrl": "192.168.1.1",
            "channel": 0,
            "messageId": "abcdefghijklmnopqrstuvwxyz123456789",
            "timestamp": 123456789,
            "sign": "abcdefghijklmnopqrstuvwxyz123456789",
            "accessory": "Meross" <You_Can_Remove_This.>
            }
            <Ends_here_from_above>
          ],
        "platform": "Meross"
        }
    ]
    ```
- Added Config for Refresh Rate.
  - default is 5 seconds and if updating to often can be set in the config.

donavanbecker committed:
Merge branch 'master' into beta
